pkg/trie/triedb/codec/decode_test.go

```diff
@@ -0,0 +1,408 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package codec
+
+import (
+	"bytes"
+	"io"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func scaleEncodeBytes(t *testing.T, b ...byte) (encoded []byte) {
+	return scaleEncodeByteSlice(t, b)
+}
+
+func scaleEncodeByteSlice(t *testing.T, b []byte) (encoded []byte) {
+	encoded, err := scale.Marshal(b)
+	require.NoError(t, err)
+	return encoded
+}
+
+func concatByteSlices(slices [][]byte) (concatenated []byte) {
+	length := 0
+	for i := range slices {
+		length += len(slices[i])
+	}
+	concatenated = make([]byte, 0, length)
+	for _, slice := range slices {
+		concatenated = append(concatenated, slice...)
+	}
+	return concatenated
```

Comment:
Use `bytes.Join` https://pkg.go.dev/bytes#Join
```suggestion
	return bytes.Join(slices, nil)
```

Reply:
The future is here! thanks for the suggestion!